packages/grpc-native-core/index.d.ts

```diff
@@ -216,18 +228,23 @@ declare module "grpc" {
     bind(port: string, creds: ServerCredentials): number;
   }
 
+  /**
+   * A type that servers as a default for an untyped service.
+   */
+  export type UntypedServiceImplementation = { [name: string]: handleCall<any, any> };
+
   /**
    * An object that completely defines a service.
    * @typedef {Object.<string, grpc~MethodDefinition>} grpc~ServiceDefinition
    */
-  export interface ServiceDefinition {
-    [s: string]: MethodDefinition;
+  export interface ServiceDefinition<ImplementationType> {
+    [s: string]: MethodDefinition<any, any>;
```

Comment:
Is it possible to use the type information in `ImplementationType` to further specify these `MethodDefinition` types?

Reply:
I tightened the definition of `ServiceDefinition`: the type can only contain keys from the `ImplementationType`. But I do not see how to apply any further typing to the `MethodDefinition`. But I think it is not too bad, `MethodDefinition` is used for run time reflection, and user code is checked against `ServiceDefinition`.